src/core1_task.rs

```diff
@@ -1111,16 +1295,56 @@ pub fn core_1_task(
             delay.delay_us(expected_rtc_sync_time_us as u32);
         }
         let one_min_check_end = timer.get_counter();
-        if frame_header_is_valid {
-            prev_frame_telemetry = Some(frame_telemetry);
-        }
-        frames_seen += 1;
+
         // info!(
         //     "Loop took {}µs, 1min check {}µs, frame transfer {}µs",
         //     (timer.get_counter() - start).to_micros(),
         //     (one_min_check_end - one_min_check_start).to_micros(),
         //     (frame_transfer_end - frame_transfer_start).to_micros()
         // );
+        if record_audio {
+            if !audio_pending && synced_date_time.date_time_utc > next_audio_alarm.unwrap() {
```

Comment:
Same comment about unwrapping `next_audio_alarm` not seeming obviously always safe to do.  Maybe do `if let Some(next_audio_alarm) = next_audio_alarm { ... }` in place of `if record_audio { ... }`?